application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -75,6 +75,7 @@
         #set($forumSpace = "$!request.space")
         #set($forumSpaceReference = $services.model.resolveSpace($forumSpace))
         #if($supportsNS &amp;&amp; $forumSpaceReference.parent.type == 'SPACE' &amp;&amp; $forumSpaceReference.parent.name == 'Forums')
+          #set ($newTopicName = "${newTopicName}_${util.date.time}_${mathtool.random(1, 99)}")
```

Comment:
This is not really the place to fix it.

The problem is in the `#displayTopicAddForm` macro, where the topic page name is initially generated. I did [mention it to you in a previous review](https://github.com/xwiki-contrib/application-forum/pull/100/files#r53635555) that you would need to re-create the `$xwiki.getUniquePageName` behavior, but in velocity, so that you end up with an unique Non-terminal page name. The original `$xwiki.getUniquePageName` is really meant for terminal documents (e.g. `space.Topic_number`) only and obviously can not be used as is for non-terminal docs (e.g. `space.Topic_number.WebHome`), like it is the case here.

You would probably need a macro that, depending if you have NS and/or if the data is migrated, either uses `$xwiki.getUniquePageName` or uses the NS-specific version to generate the page name and use it in the `#displayTopicAddForm` macro.

At the current level (in Actions.xml) you should not care about this.